lib/AST/Type.cpp

```diff
@@ -4109,6 +4109,62 @@ bool Type::isOrContainsUncheckedType() const {
   }
 }
 
+// containsCheckedValue - check whether a field type is a checked type or is a 
+// constructed type (array, pointer, function) that uses a checked type.
+bool Type::containsCheckedValue() const {
+  const Type *current = CanonicalType.getTypePtr();
+  switch (current->getTypeClass()) {
+  case Type::Pointer: {
+    const PointerType *ptr = cast<PointerType>(current);
+    if (ptr->isCheckedPointerType()) {
+      return true;
+    }
+    return false;
+  }
+  case Type::ConstantArray:
+  case Type::DependentSizedArray:
+  case Type::IncompleteArray:
+  case Type::VariableArray: {
+
+    return current->getPointeeOrArrayElementType()->containsCheckedValue();
+  }
+
+  case Type::FunctionProto: {

```

Comment:
Function types should not be included.   We are checking whether a checked pointer value is stored in an object in memory.  A function type may have a checked pointer type as an argument or a return value.  That doesn't mean that a checked pointer value is stored in an object in memory.